enh(snmp connectors): add native SNMP v3 macros CTOR-2360 (#5400)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/network-switchs-hp-procurve-snmp.md

| Service Alias | Service Template                        | Service Description                                             |
|:--------------|:----------------------------------------|:----------------------------------------------------------------|
| Cpu           | Net-Hp-Procurve-Cpu-SNMP-custom         | Check the rate of utilization of CPU for the machine            |

> | Environment   | Net-Hp-Procurve-Environment-SNMP-custom | Check hardware environment (Fans, Power Supplies, Temperatures) |

Author: ```suggestion
| Cpu           | Net-Hp-Procurve-Cpu-SNMP-custom         | Check the rate of CPU utilization for the machine            |
| Environment   | Net-Hp-Procurve-Environment-SNMP-custom | Check hardware environment (fans, power supplies, temperatures) |
```

</TabItem>
<TabItem value="Not attached to a host template" label="Not attached to a host template">

| Service Alias        | Service Template                                 | Service Description                                                    | Discovery |
|:---------------------|:-------------------------------------------------|:-----------------------------------------------------------------------|:---------:|
| Spanning-Tree        | Net-HP-Procurve-SpanningTree-SNMP-custom         | Monitor the status of the Spanning Tree Protocol on network interfaces | X         |
| Stack                | Net-Hp-Procurve-Stack-SNMP-custom                | Check stack members                                                    |           |
| Traffic-Generic-Id   | Net-Hp-Procurve-Traffic-Id-Generic-SNMP-custom   | Check traffic of an network interface                                  |           |

> | Traffic-Generic-Name | Net-Hp-Procurve-Traffic-Name-Generic-SNMP-custom | Check traffic of an network interface                                  |           |

Author: ```suggestion
| Traffic-Generic-Id   | Net-Hp-Procurve-Traffic-Id-Generic-SNMP-custom   | Check traffic of a network interface                                  |           |
| Traffic-Generic-Name | Net-Hp-Procurve-Traffic-Name-Generic-SNMP-custom | Check traffic of a network interface                                  |           |
```

|:------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| SNMP Agents | Discover your resources through an SNMP subnet scan. You need to install the [Generic SNMP](./applications-protocol-snmp.md) connector to get the discovery rule and create a template mapper for the **Net-Hp-Procurve-SNMP-custom** host template. |

More information about discovering hosts automatically is available on the [dedicated page](/docs/monitoring/discovery/hosts-discovery).

#### Service discovery

| Rule name                               | Description                                                                                  |
|:----------------------------------------|:---------------------------------------------------------------------------------------------|
| Net-Hp-Procurve-SNMP-Spanning-Tree-Name | Discover network interfaces and monitor bandwidth utilization                                |

> | Net-Hp-Procurve-SNMP-Traffic-Name       | Discover the network interfaces in order to monitor the status of the Spanning Tree Protocol |

Author: ```suggestion
| Net-Hp-Procurve-SNMP-Spanning-Tree-Name | Discover the network interfaces in order to monitor the status of the Spanning Tree Protocol |
| Net-Hp-Procurve-SNMP-Traffic-Name       | Discover network interfaces and monitor bandwidth utilization                                |
```

<Tabs groupId="sync">
<TabItem value="Cpu" label="Cpu">

| Name      | Unit |
|:----------|:-----|
| cpu_usage | %    |

</TabItem>
<TabItem value="Environment" label="Environment">

| Name                       | Unit  |
|:---------------------------|:------|
| hardware.fan.count         | count |
| fan status                 | N/A   |
| hardware.psu.count         | count |
| psu status                 | N/A   |
| hardware.sensor.count      | count |
| sensor status              | N/A   |
| hardware.temperature.count | count |
| temperature status         | N/A   |

</TabItem>
<TabItem value="Memory" label="Memory">

| Name                 | Unit |
|:---------------------|:-----|
| used_*instance_name* | B    |
</TabItem>
<TabItem value="Spanning-Tree" label="Spanning-Tree">

| Name   | Unit |
|:-------|:-----|
| status | N/A  |

> To obtain this new metric format, include **--use-new-perfdata** in the **EXTRAOPTIONS** service macro.

</TabItem>
<TabItem value="Stack" label="Stack">

| Name          | Unit |
|:--------------|:-----|
| member-status | N/A  |
| port-status   | N/A  |

</TabItem>
<TabItem value="Traffic-*" label="Traffic-*">

> Coming soon

Author: ```suggestion
| Name                                                 | Unit |
|:-----------------------------------------------------|:-----|
| *interface_name*#status                              | N/A  |
| *interface_name*#interface.traffic.in.bitspersecond  | b/s  |
| *interface_name*#interface.traffic.out.bitspersecond | b/s  |
```

1. Log into Centreon and add a new host through **Configuration > Hosts**.
2. Fill in the **Name**, **Alias** & **IP Address/DNS** fields according to your resource's settings.
3. Apply the **Net-Hp-Procurve-SNMP-custom** template to the host.

> When using SNMP v3, use the **SNMPEXTRAOPTIONS** macro to add specific authentication parameters.

> > More information in the [Troubleshooting SNMP](../getting-started/how-to-guides/troubleshooting-plugins.md#snmpv3-options-mapping) section.

Author: ```suggestion
```

```bash
/usr/lib/centreon/plugins/centreon_hp_procurve.pl \
	--plugin=network::hp::procurve::snmp::plugin \
	--mode=spanning-tree \
	--hostname=10.0.0.1 \
	--snmp-version='2c' \
	--snmp-community='my-snmp-community' \

> 	--snmp-username='username' \

Author: ```suggestion
	--snmp-username='' \
```
